### PR TITLE
Add new method 'getlist' to 'ConfigParser'

### PR DESCRIPTION
**Pull request by SpotlightKid**
This allows to get a configuration value as a list of strings.

For now this mainly to support PR #113, but could be used by future new features as well.

Here's a copy of the included docstring:

```
Return value of option in given section as a list of strings.

If option is not set, return default (defaults to an empty list).

The option value is split into list tokens using one of two strategies:

* If the value contains any newlines, i.e. it was written in the
  configuration file using continuation lines, the value is split at
  newlines and empty items are discarded.
* Otherwise, the value is split according to unix shell parsing rules.
  Items are separated by whitespace, but items can be enclosed in 
  single or double quotes to preserve spaces in them.
  
Example::

    [test]
    option2 =
        one
        two three
        four
        sive six
    option1 = one  "two three" four 'five  six'
```

### CODE REVIEW

**Comment by k4nar** [comment]
LGTM :+1: 

It only adds what's needed know, but it can easily be extended in the future, I like it.

File: watson/config.py [hunk 1]
```diff
@@ -60,6 +62,42 @@ def getboolean(self, section, option, default=False):
...
+        """
+        Return value of option in given section as a list of strings.
+
+        If option is not set, return default (defaults to an empty list).
+
+        The option value is split into list tokens using one of two strategies:
+
+        * If the value contains any newlines, i.e. it was written in the
+          configuration file using continuation lines, the value is split at
+          newlines and empty items are discarded.
```
Comment (jbarnoud) [inline watson/config.py]: It may be worth having a test with en empty line to make sure it is discarded.

Comment (SpotlightKid) [inline watson/config.py]: The test already covers that with `value1` in the test config, where the first option line is empty. But I will add another empty line on its own, to demonstrate that this works too.

**Comment by SpotlightKid** [comment]
Is anything preventing this from getting merged? Finalizing & merging PR #113 depends on it.

File: watson/config.py [hunk 2]
```diff
@@ -60,6 +62,42 @@ def getboolean(self, section, option, default=False):
...
+            option1 = one  "two three" four 'five  six'
+
+        """
+        if not self.has_option(section, option):
+            return [] if default is None else default
+
+        value = self.get(section, option)
+
+        if '\n' in value:
+            return [item for item in value.splitlines() if item]
```
Comment (k4nar) [inline watson/config.py]: shouldn't we strip each item ? Looking at the tests it seems to work that way, but I'm not sure to understand why.

**Comment by k4nar** [comment]
Sorry about the delay :( .

I made a small comment because there is something that I don't quite understand, but otherwise I'll merge it.

File: watson/config.py (hunk 2 above)
Comment (SpotlightKid) [inline watson/config.py]: Per the [configparser docs](https://docs.python.org/2/library/configparser.html) leading whitespace is stripped from values.

I agree it's probably best to explicitly strip items in line 97 to also strip trailing whitespace.

Just say the word and I'll update the PR.

Comment (k4nar) [inline watson/config.py]: Yeah I think this would be a little bit clearer :) .

**Comment by SpotlightKid** [comment]
I've updated the PR to strip list item values when continuation lines are used.
